src/utils/builder.ts

```diff
@@ -69,46 +69,131 @@ export function setCriteriaElm(pathway: Pathway, elm: object): void {
   pathway.elm.criteria = elm;
 }
 
-function addState(pathway: Pathway, key?: string): string {
+export function createState(key?: string): State {
   if (!key) key = shortid.generate();
   const state: State = {
-    key: key,
-    label: '',
+    key,
+    label: 'New Node',
     transitions: []
   };
-  pathway.states[key] = state;
 
-  return key;
+  return state;
 }
 
-export function addBranchState(pathway: Pathway): string {
-  const key = shortid.generate();
-  return addState(pathway, key);
+export function addState(pathway: Pathway, state: State): Pathway {
+  return {
+    ...pathway,
+    states: {
+      ...pathway.states,
+      [state.key as string]: state
+    }
+  };
 }
 
-export function addGuidanceState(pathway: Pathway): string {
-  const key = shortid.generate();
-  addState(pathway, key);
-  makeBranchStateGuidance(pathway, key);
-  return key;
+export function addGuidanceState(pathway: Pathway): Pathway {
+  const state = createState();
+  const newPathway = addState(pathway, state);
+
+  return makeBranchStateGuidance(newPathway, state.key as string);
 }
 
-export function setStateLabel(pathway: Pathway, key: string, label: string): void {
-  pathway.states[key].label = label;
+export function setStateLabel(pathway: Pathway, key: string, label: string): Pathway {
+  return {
+    ...pathway,
+    states: {
+      ...pathway.states,
+      [key]: {
+        ...pathway.states[key],
+        label
+      }
+    }
+  };
+}
+
+export function setStateNodeType(pathway: Pathway, key: string, nodeType: string): Pathway {
+  return {
+    ...pathway,
+    states: {
+      ...pathway.states,
+      [key]: {
+        ...pathway.states[key],
+        nodeType
+      }
+    }
+  };
+}
+
+export function setStateCriteriaSource(
+  pathway: Pathway,
+  key: string,
+  criteriaSource: string
+): Pathway {
+  return {
+    ...pathway,
+    states: {
+      ...pathway.states,
+      [key]: {
+        ...pathway.states[key],
+        criteriaSource
```

Comment:
CriteriaSource was originally left off the interface because the Camino app does not support it

Reply:
Yeah this will go away with the v2 designs. I left it in because it was in the v1 design and implemented with this task.

Reply:
V2 being [PATHWAYS-259](https://jira.mitre.org/browse/PATHWAYS-259)?